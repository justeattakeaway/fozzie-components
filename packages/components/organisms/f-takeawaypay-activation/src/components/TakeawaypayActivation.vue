<template>
    <div
        :class="$style['c-takeawaypayActivation']"
        data-test-id="takeawaypayActivation">
        {{ copy.text }}
    </div>
</template>

<script>
import { globalisationServices } from '@justeat/f-services';
import tenantConfigs from '../tenants';

export default {
    props: {
        locale: {
            type: String,
            default: ''
        }
    },
    data () {
        const locale = globalisationServices.getLocale(tenantConfigs, this.locale, this.$i18n);
        const localeConfig = tenantConfigs[locale];

        return {
            copy: { ...localeConfig }
        };
    }
};
</script>

<style lang="scss" module>

.c-takeawaypayActivation {
    display: flex;
    justify-content: center;
    min-height: 80vh;
    width: 80vw;
    margin: auto;
    font-family: $font-family-base;
    @include font-size(heading-m);
}

</style>
