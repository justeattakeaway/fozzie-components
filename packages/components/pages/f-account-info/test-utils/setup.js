import Vuex from 'vuex';
import { createLocalVue } from '@vue/test-utils';
import { VueI18n } from '@justeat/f-globalisation';
import tenantConfigs from '../src/tenants';

export const localVue = createLocalVue();
localVue.use(VueI18n);
localVue.use(Vuex);

export const i18n = {
    locale: 'en-GB',
    messages: {
        'en-GB': tenantConfigs['en-GB'].messages
    }
};

export const baseUrl = 'https://smartGatewayBaseUrl.com';

export const token = 'some-auth-token';

export const conversationId = 'b7386108-95e6-4e73-9421-5b066c089153';

export const consumerDetailsGetResponse = {
    data: {
        ConsumerId: 1050450174,
        Id: '8It2lEaJjG82nJtxO00Iab0CoF0=',
        CreatedDate: '2019-04-16T17:25:00.000Z',
        EmailAddress: 'satchin.moriyoka@example.com',
        FirstName: 'Satchin',
        LastName: 'Moriyoka',
        PhoneNumber: '+441234567890',
        DateOfBirth: '1914-04-16T17:25:00.000Z',
        TermsAndConditionsId: 0,
        WantsNewsletter: true,
        AccountType: 'Default',
        ConsumerStatus: 'ReturnConsumer'
    }
};

const consumerAddressGetResponse = {
    AddressId: 1050450174,
    City: 'Bristol',
    ZipCode: 'BS1 4DJ',
    AddressName: 'Office',
    IsDefault: true, // Key field
    Line1: 'Just Eat',
    Line2: 'Broad Quay House'
};

export const consumerAddressesGetResponse = {
    data: [{
        ...consumerAddressGetResponse
    },
    {
        AddressId: 1050450175,
        City: 'Bristol',
        ZipCode: 'BS1 9DJ',
        AddressName: 'Office',
        IsDefault: false,
        Line1: 'Takeaway',
        Line2: 'Broad Quay House'
    }]
};

export const consumerDetailsMappedModel = {
    id: consumerDetailsGetResponse.data.ConsumerId,
    firstName: consumerDetailsGetResponse.data.FirstName,
    lastName: consumerDetailsGetResponse.data.LastName,
    phoneNumber: consumerDetailsGetResponse.data.PhoneNumber,
    emailAddress: consumerDetailsGetResponse.data.EmailAddress
};

export const consumerAddressMappedModel = {
    line1: consumerAddressGetResponse.Line1,
    line2: consumerAddressGetResponse.Line2,
    line3: consumerAddressGetResponse.Line3,
    locality: consumerAddressGetResponse.City,
    postcode: consumerAddressGetResponse.ZipCode
};

export const consumerViewModel = {
    ...consumerDetailsMappedModel,
    ...consumerAddressMappedModel
};

export const consumerStateModel = {
    consumer: {
        ...consumerDetailsMappedModel,
        ...consumerAddressMappedModel
    }
};
