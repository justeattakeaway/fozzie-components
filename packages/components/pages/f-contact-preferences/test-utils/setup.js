export const contactPreferencesGetResponse = {
    data: {
        Preferences: [{
            DisplayName: 'Order status',
            Sort: 1,
            Key: 'orderstatus',
            Push: false,
            Email: true,
            Sms: true
        }, {
            DisplayName: 'Review meal',
            Sort: 2,
            Key: 'reviewmeal',
            Push: false,
            Email: false,
            Sms: false
        }, {
            DisplayName: 'News & offers',
            Sort: 3,
            Key: 'news',
            Push: false,
            Email: false,
            Sms: false
        }
        ],
        PreferenceVersionViewed: 0
    }
};

export const contactPreferencesViewModel = {
    preferenceVersionViewed: 0,
    preferences: [{
        displayName: 'Order status',
        isEmailEnabled: false,
        emailValue: true,
        key: 'orderstatus',
        pushEnabled: false,
        pushValue: false,
        isSmsEnabled: false,
        smsValue: true,
        sort: 1,
        visible: true
    }, {
        displayName: 'Review meal',
        isEmailEnabled: true,
        emailValue: false,
        key: 'reviewmeal',
        pushEnabled: true,
        pushValue: false,
        isSmsEnabled: true,
        smsValue: false,
        sort: 2,
        visible: false
    }, {
        displayName: 'News & offers',
        isEmailEnabled: true,
        emailValue: false,
        key: 'news',
        pushEnabled: true,
        pushValue: false,
        isSmsEnabled: true,
        smsValue: false,
        sort: 3,
        visible: true
    }]
};

export const filteredContactPreferencesModel = [{
    displayName: 'Order status',
    isEmailEnabled: false,
    emailValue: true,
    key: 'orderstatus',
    pushEnabled: false,
    pushValue: false,
    isSmsEnabled: false,
    smsValue: true,
    sort: 1,
    visible: true
}, {
    displayName: 'News & offers',
    isEmailEnabled: true,
    emailValue: false,
    key: 'news',
    pushEnabled: true,
    pushValue: false,
    isSmsEnabled: true,
    smsValue: false,
    sort: 3,
    visible: true
}];

export const contactPreferencesUpdateModel = {
    Preferences: [{
        DisplayName: 'Order status',
        Sort: 1,
        Key: 'orderstatus',
        Push: false,
        Email: true,
        Sms: true
    }, {
        DisplayName: 'Review meal',
        Sort: 2,
        Key: 'reviewmeal',
        Push: false,
        Email: false,
        Sms: false
    }, {
        DisplayName: 'News & offers',
        Sort: 3,
        Key: 'news',
        Push: false,
        Email: false,
        Sms: false
    }],
    DeviceToken: null,
    DeviceType: null,
    PhoneNumber: null,
    PreferenceVersionViewed: 0
};

export const baseUrl = 'https://smartGatewayBaseUrl.com';

export const token = 'some-auth-token';

export const conversationId = 'b7386108-95e6-4e73-9421-5b066c089153';

export const contactPreferencesUpdateBody = {
    Preference: [{
        DisplayName: 'Order status',
        Sort: 1,
        Key: 'orderstatus',
        Push: false,
        Email: true,
        Sms: true
    }, {
        DisplayName: 'Review meal',
        Sort: 2,
        Key: 'reviewmeal',
        Push: false,
        Email: false,
        Sms: false
    }, {
        DisplayName: 'News & offers',
        Sort: 3,
        Key: 'news',
        Push: false,
        Email: true,
        Sms: false
    }
    ],
    DeviceToken: null,
    DeviceType: null,
    PhoneNumber: null,
    PreferenceVersionViewed: 0
};
