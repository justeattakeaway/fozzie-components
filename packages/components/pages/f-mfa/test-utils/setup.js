export const baseUrl = 'https://smartGatewayBaseUrl.com';

export const conversationId = 'b7386108-95e6-4e73-9421-5b066c089153';

// eslint-disable-next-line camelcase
export const postBody = { mfa_token: 'some_mfa_token', otp: 'ABC123' };
