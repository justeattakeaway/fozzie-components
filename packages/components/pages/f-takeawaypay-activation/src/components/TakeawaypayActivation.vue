<template>
    <div :class="$style['c-takeawaypayActivation']">
        <card-component
            is-page-content-wrapper
            card-heading-position="center"
            data-test-id="takeawaypay-activation-component"
            :class="$style['c-takeawaypayActivation-card']">
            <div
                v-if="shouldShowSpinner"
                :class="$style['c-spinner-wrapper']"
                data-test-id="takeawaypay-loading-spinner">
                <div :class="$style['c-spinner']" />
            </div>

            <component
                :is="activationStateComponent.name"
                v-if="!shouldShowSpinner"
                ref="activationStateComponent"
                v-bind="activationStateComponent.props"
                @activation-result="handleActivationResult" />
        </card-component>
    </div>
</template>

<script>
import jwtDecode from 'jwt-decode';
import { VueGlobalisationMixin } from '@justeat/f-globalisation';
import CardComponent from '@justeat/f-card';
import '@justeat/f-card/dist/f-card.css';
import tenantConfigs from '../tenants';
import ActivationLoggedIn from './ActivationLoggedIn.vue';
import ActivationNotLoggedIn from './ActivationNotLoggedIn.vue';
import ActivationSuccessful from './ActivationSuccessful.vue';
import ActivationFailed from './ActivationFailed.vue';
import TakeawaypayActivationServiceApi from '../services/TakeawaypayActivationServiceApi';
import {
    ACTIVATION_STATE_NONE,
    ACTIVATION_STATE_AVAILABLE_LOGGED_IN,
    ACTIVATION_STATE_AVAILABLE_NOT_LOGGED_IN,
    ACTIVATION_STATE_SUCCEEDED,
    ACTIVATION_STATE_FAILED,
    USER_ROLE_GUEST
} from '../constants';

export default {

    components: {
        CardComponent,
        ActivationLoggedIn,
        ActivationNotLoggedIn,
        ActivationSuccessful,
        ActivationFailed
    },

    mixins: [
        VueGlobalisationMixin
    ],

    props: {
        getActivationStatusUrl: {
            type: String,
            required: true
        },
        activateUrl: {
            type: String,
            required: true
        },
        loginUrl: {
            type: String,
            required: true
        },
        registrationUrl: {
            type: String,
            required: true
        },
        homeUrl: {
            type: String,
            required: true
        },
        authToken: {
            type: String,
            default: ''
        },
        employeeId: {
            type: String,
            default: ''
        }
    },
    data () {
        return {
            tenantConfigs,
            consumerId: '',
            consumerEmail: '',
            consumerGivenName: '',
            consumerRole: '',
            activationState: ACTIVATION_STATE_NONE,
            shouldShowSpinner: false
        };
    },

    computed: {
        activationStateComponent () {
            if (this.activationState === ACTIVATION_STATE_AVAILABLE_LOGGED_IN) {
                return this.activationLoggedIn;
            }

            if (this.activationState === ACTIVATION_STATE_AVAILABLE_NOT_LOGGED_IN) {
                return this.activationNotLoggedIn;
            }

            if (this.activationState === ACTIVATION_STATE_SUCCEEDED) {
                return this.activationSucceeded;
            }
            return this.activationFailed;
        },

        activationFailed () {
            return {
                name: 'activation-failed',
                props: {
                    redirectUrl: this.redirectUrl
                }
            };
        },

        activationSucceeded () {
            return {
                name: 'activation-successful',
                props: {
                    homeUrl: this.homeUrl
                }
            };
        },

        activationLoggedIn () {
            return {
                name: 'activation-logged-in',
                props: {
                    loginUrl: this.loginUrl,
                    registrationUrl: this.registrationUrl,
                    activateUrl: this.activateUrl,
                    authToken: this.authToken,
                    consumerId: this.consumerId,
                    consumerEmail: this.consumerEmail,
                    consumerGivenName: this.consumerGivenName,
                    employeeId: this.employeeId
                }
            };
        },

        activationNotLoggedIn () {
            return {
                name: 'activation-not-logged-in',
                props: {
                    loginUrl: this.loginUrl,
                    registrationUrl: this.registrationUrl
                }
            };
        }
    },

    async mounted () {
        await this.initialize();
    },

    methods: {
        async initialize () {
            this.shouldShowSpinner = true;
            const available = await TakeawaypayActivationServiceApi.isActivationAvailable(this.getActivationStatusUrl, this.employeeId, this.$store, this.$logger);

            if (available) {
                this.determineActivationState();
            } else {
                this.activationState = ACTIVATION_STATE_FAILED;
            }
            this.shouldShowSpinner = false;
        },

        extractConsumerDetails () {
            const tokenData = jwtDecode(this.authToken);
            this.consumerId = tokenData.sub;
            this.consumerEmail = tokenData.email;
            this.consumerGivenName = tokenData.given_name;
            this.consumerRole = (tokenData.role || '').toLowerCase();
        },

        determineActivationState () {
            if (this.authToken) {
                this.extractConsumerDetails();
                this.activationState = this.consumerRole === USER_ROLE_GUEST ?
                    ACTIVATION_STATE_AVAILABLE_NOT_LOGGED_IN :
                    ACTIVATION_STATE_AVAILABLE_LOGGED_IN;
            } else {
                this.activationState = ACTIVATION_STATE_AVAILABLE_NOT_LOGGED_IN;
            }
        },

        handleActivationResult (successful) {
            this.activationState = successful ? ACTIVATION_STATE_SUCCEEDED : ACTIVATION_STATE_FAILED;
        }
    }
};
</script>

<style lang="scss" module>

@include loadingIndicator('large');

.c-spinner-wrapper {
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);

    .c-spinner {
        margin: 0 auto;
        border: 3px solid $color-content-brand;
        border-top: 3px solid rgba(243, 109, 0, 0.2);
    }
}

.c-takeawaypayActivation {
    display: flex;
    justify-content: center;
    min-height: 80vh;
    margin: auto;
    font-family: $font-family-base;
    @include font-size(heading-m);
    text-align: center;

    @include media('>=narrow') {
        width: 80vw;
    }
}

    .c-takeawaypayActivation-card {
        position: relative;

        @include media('>=narrow') {
            box-shadow: 0 1px 1px 0 rgba($color-black, 0.03), 0 2px 1px -1px rgba($color-black, 0.07), 0 1px 3px 0 rgba($color-black, 0.06);
        }
    }

</style>
