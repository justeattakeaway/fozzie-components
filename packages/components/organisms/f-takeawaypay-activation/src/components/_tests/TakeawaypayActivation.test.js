import { shallowMount } from '@vue/test-utils';
import TakeawaypayActivation from '../TakeawaypayActivation.vue';

describe('TakeawaypayActivation', () => {
    it('should be defined', () => {
        const propsData = {};
        const wrapper = shallowMount(TakeawaypayActivation, { propsData });
        expect(wrapper.exists()).toBe(true);
    });
});
