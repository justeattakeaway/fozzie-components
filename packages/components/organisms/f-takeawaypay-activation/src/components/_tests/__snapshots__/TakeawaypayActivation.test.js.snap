// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TakeawaypayActivation should render ActivationFailed component 1`] = `
<card-component-stub cardheading="" cardheadingposition="center" cardheadingtag="h1" ispagecontentwrapper="true" isrounded="true" data-test-id="takeawaypay-activation-component">
  <activation-failed-stub></activation-failed-stub>
</card-component-stub>
`;

exports[`TakeawaypayActivation should render ActivationLoggedIn component 1`] = `
<card-component-stub cardheading="" cardheadingposition="center" cardheadingtag="h1" ispagecontentwrapper="true" isrounded="true" data-test-id="takeawaypay-activation-component">
  <activation-logged-in-stub loginurl="/login" registrationurl="/register" activateurl="/activate" authtoken="token" consumerid="" consumeremail=""></activation-logged-in-stub>
</card-component-stub>
`;

exports[`TakeawaypayActivation should render ActivationNotLoggedIn component 1`] = `
<card-component-stub cardheading="" cardheadingposition="center" cardheadingtag="h1" ispagecontentwrapper="true" isrounded="true" data-test-id="takeawaypay-activation-component">
  <activation-not-logged-in-stub loginurl="/login" registrationurl="/register"></activation-not-logged-in-stub>
</card-component-stub>
`;

exports[`TakeawaypayActivation should render ActivationSuccessful component 1`] = `
<card-component-stub cardheading="" cardheadingposition="center" cardheadingtag="h1" ispagecontentwrapper="true" isrounded="true" data-test-id="takeawaypay-activation-component">
  <activation-successful-stub homeurl="/"></activation-successful-stub>
</card-component-stub>
`;
